src/backend/google-auth.js

```diff
@@ -0,0 +1,156 @@
+require('dotenv').config();
+import express from 'express';
+import db from './db';
+import { OAuth2Client, JWT } from 'google-auth-library';
+import {google} from 'googleapis'
+
+const client = new OAuth2Client(process.env.GOOGLE_CLIENT_ID);
+
+const router = express.Router();
+
+router.post('/', async (req, res) => {
+  const { tokenId } = req.query;
+  if (!tokenId) {
+    res.send('Missing Token ID').status(400);
+    return;
+  }
+  try {
+    const ticket = await client.verifyIdToken({
+      idToken: tokenId,
+      audience: process.env.GOOGLE_CLIENT_ID,
+    })
+    const payload = ticket.getPayload();
+    const {
+      hd, // host domain
+      sub: userId,
+      email,
+      family_name,
+      given_name,
+    } = payload;
+
+    // check if user is a member of waterloop
+    if (hd !== 'waterloop.ca') {
+      throw {
+        status: 403,
+        msg: 'Attempted Sign in from not waterloop.ca account',
+      };
+    }
+
+    // check if user is a lead
+    const lead = await isLead(email)
+    if (!lead) {
+      throw {
+        status: 403,
+        msg: `User email ${email} is not a member of the Leads group`,
+      };
+    }
+
+    // get user info
+    const userQueryResp = await db.users.getById(userId)
+
+    if (userQueryResp === -1) {
+      // No user found so make one
+      await db.users.createUser(email, given_name, family_name, userId)
+      await db.users.getById(userId)
```

Comment:
What's the purpose of this getby statement ?

Reply:
Def looks like we can remove it :)